aiidalab_widgets_base/computers.py

```diff
@@ -693,14 +698,18 @@ def __init__(self, text='Select computer:', path_to_root='../', **kwargs):
         self._refresh()
         super().__init__(children=children, **kwargs)
 
-    @staticmethod
-    def _get_computers():
+    def _get_computers(self):
         """Get the list of available computers."""
-        query_b = QueryBuilder()
-        query_b.append(Computer, tag='computer')
+        querybuild = QueryBuilder()
+        querybuild.append(Computer, tag='computer')
+
+        if self.disabled_computers:
+            to_return = [c[0] for c in querybuild.all()]
+        else:
+            to_return = [c[0] for c in querybuild.all() if c[0].is_user_enabled(User.objects.get_default())]
 
         # Only computers configured for the current user.
-        return {c[0].name: c[0] for c in query_b.all() if c[0].is_user_configured(User.objects.get_default())}
+        return {c.name: c for c in to_return if c.is_user_configured(User.objects.get_default())}
```

Comment:
I think we can simplify this segment a little bit by converting this into one comprehension (with appropriate formatting of course):
```python
user = User.objects.get_default()
return {
    c[0].name: c[0] for c in query_b.all()
    if c[0].is_user_configured(user) and (self.disabled_computers or c[0].is_user_enabled(user)}
```

Reply:
thanks, I did that.